.github/workflows/ci-all-in-one-build.yml

```diff
@@ -11,19 +11,24 @@ jobs:
   all-in-one:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
+    - name: Harden Runner
+      uses: step-security/harden-runner@2e205a28d0e1da00c5f53b161f4067b052c61f34
```

Comment:
https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions > Indirectly yes they suggested this, otherwise you have to do all of this work yourself manually. We could do that approach.